### DISCUSSION TITLE
Windows installation

### DISCUSSION THREAD

**Discussion by olivierLA75** [guydavis/machinaris#229] [Q&A] [answered]
Hey Guy, I apologize for all my questions and I should be done soon. I'm trying to install Machinaris on Windows 10. I installed docker but when I paste the docker run command in Command Prompt it doesn't work at all and ives a bunch of errors. Any idea what's the issue? (see screenshot) I tried in Powershell and what similar issue
<img width="705" alt="Screen Shot 2021-08-17 at 5 23 43 PM" src="https://user-images.githubusercontent.com/83734065/129817407-aeddeb03-8d44-4c7c-b3ff-4307be8136fe.png">

thanks
Olivier

**Comment by guydavis** [comment]
Please post same screen showing PowerShell. Thanks!

**Comment by olivierLA75** [comment]
<img width="835" alt="Screen Shot 2021-08-17 at 5 31 29 PM" src="https://user-images.githubusercontent.com/83734065/129817887-57869805-7475-4435-8b3d-2cf1096a78ca.png">

**Reply by guydavis** [reply]
Hi, sorry for the trouble.  In Powershell, try again, but this time remove all the double-quotes in the command in a text editor first.

**Comment by olivierLA75** [comment]
<img width="835" alt="Screen Shot 2021-08-17 at 5 39 34 PM" src="https://user-images.githubusercontent.com/83734065/129818539-3a6b3798-6224-4af0-b88a-ad2a12e0b53e.png">

**Reply by guydavis** [reply]
Sorry, I see the error now.  The issue is the "Purple Rabbit" path with a space in it.  I think on Windows, you need to backslash escape it like this:
`-v X:/Purple\ Rabbit:/plots13`
and your **other volume paths** with spaces in the name.
Please experiment to see if you can escape them in a way that Docker allows.  Else, I suggest renaming your volumes without spaces.
Hope this helps.

**Reply by olivierLA75** [reply]
so I changed the folder names and removed the spaces, tried both with and without the double-quotes but still the same error
<img width="846" alt="Screen Shot 2021-08-17 at 6 01 32 PM" src="https://user-images.githubusercontent.com/83734065/129820072-f0eaec91-7c41-4ca5-bf89-7d9c332e56b5.png">

**Reply by guydavis** [reply]
Okay, so I've now tested this on my own Windows 10 test VM, using a handful of paths with spaces:
![image](https://user-images.githubusercontent.com/12936/129917451-e50934a9-ee65-4535-b325-937ee8f06697.png)

There was no problem with the volumes containing spaces when double-quoted, same as the Launch Wizard currently does.  

So, this makes me think there is something wrong with the networking configuration on your LAN.  Please try to run the original Docker run command, with original double-quoting again.  However, this time, remove the `--dns ...` line entirely to bypass using your LAN's local DNS.  It seems that your Docker is having trouble pulling an image from the Github Container Registry.  64.71.0.34 may not be the correct IP for ghcr.io.

If that doesn't work, you can try pulling the Machinaris image from the Docker Registry instead, by replacing line: 
```
ghcr.io/guydavis/machinaris
```
with 
```
guydavis/machinaris
```
This should attempt to pull the image from Docker Registry instead, which may work better from your network location.  Let me know how it goes.

**Comment by olivierLA75** [comment, marked as answer]
it worked. the DNS was the issue. thank you so much for your help!
